node.go

```diff
@@ -276,6 +282,11 @@ func (n *node) emittedCount() (count int64) {
 	return
 }
 
+// node increment error count increments a nodes error_count stat
+func (n *node) incrementErrorCount() {
+	n.statMap.Add(statErrorCount, 1)
```

Comment:
To avoid the map look up each time you should keep the `errCountVar` from above on the node struct and then just increment it directly.